client/login/ssr.js

```diff
@@ -65,3 +67,29 @@ export function ssrSetupLocaleLogin( context, next ) {
 
 	next();
 }
+
+export function setMetaTags( context, next ) {
+	const pathSegments = context.pathname.replace( /^[/]|[/]$/g, '' ).split( '/' );
+	const hasQueryString = Object.keys( context.query ).length > 0;
+	const hasMag16LocaleParam = config( 'magnificent_non_en_locales' ).includes(
+		context.params?.lang
+	);
+
+	/**
+	 * Only the main `/log-in` and `/log-in/[mag-16-locale]` routes should be indexed.
+	 */
+	if ( hasQueryString || pathSegments.length > ( hasMag16LocaleParam ? 2 : 1 ) ) {
```

Comment:
This is all getting repetitive (with #95013 for example). Have we considered adding some sort of helper to reuse that logic, and maybe add some tests to ensure it works (and continues to work) correctly?

Reply:
I agree - we plan on addressing that in a follow-up PR later this week.